scripts/pylib/twister/twisterlib/handlers.py

```diff
@@ -694,7 +694,10 @@ def handle(self, harness):
             timeout = 30
             if script_param:
                 timeout = script_param.get("pre_script_timeout", timeout)
-            self.run_custom_script(pre_script, timeout)
+            pre_script_cmd = ([pre_script] +
```

Comment:
what if with this change we also allow these `*script` properties to convey a script command line with parameters,
not just a script file path as it is now, but doing like this 
```
pre_script_cmd = (pre_script.split() +
                              [self.instance.name]
                             )
```

Reply:
Maybe instead of passing the instance name every time to those scripts, better to allow user to add custom parameters in hardware map, e.g.
`pre_script: '<PATH>/script.py --input platform_name'`
and better to modify `run_custom_script` method:
- just add `shell=True` to subprocess.Popen
- or use `shlex.split(script)` to safely split input parameters (to avoid issues with shell injections)

If you add additional parameter to every call, you definitely must to document it (as @golowanow requested), because some of downstream users can use custom scripts that do not allow additional parameters

Reply:
1. platform name is not enough, most of time we need to know the testinstance name to decide how to run pre_script.
2. shell-True sometime does not work well in a cloud system.

Reply:
> most of time we need to know the testinstance name to decide how to run pre_script.

I still don't think it is good idea to pass this info every time to the pre/post scripts. If you need something extra in your internal scripts, you can just read it from build folder (name of directory consists a platform name and test name ... or just take it from CMakeCache.txt or other build artefact).

> shell-True sometime does not work well in a cloud system.

try to use split to pass a list to subprocess

Reply:
if we have 8 instance running, how do I know which one is the current directory to parse?

Reply:
> I still don't think it is good idea to pass this info every time to the pre/post scripts. If you need something extra in your internal scripts, you can just read it from build folder (name of directory consists a platform name and test name ... or just take it from CMakeCache.txt or other build artefact).

By parsing the current build system to get the instance name,  this costs more. and the instance name is so far the only thing we need to identify, as we know the platform + instance_name, it will uniquely identify the test case context.